snapshots.nix

```diff
@@ -0,0 +1,57 @@
+{ lib, mkPkgSet, stackage }:
```

Comment:
Can we have a header comment here that explains the motivation behind `snapshot.nix` a bit?

Reply:
Yep, good idea, fixed.